File: app/src/main/java/com/surdroid/networkchangedetect/NetworkUtil.java
package com.surdroid.networkchangedetect;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;
import android.net.wifi.WifiConfiguration;
import android.net.wifi.WifiManager;

import java.util.ArrayList;
import java.util.List;

public class NetworkUtil {

    public static int TYPE_WIFI = 1;
    public static int TYPE_MOBILE = 2;
    public static int TYPE_NOT_CONNECTED = 0;


    public static int getConnectivityStatus(Context context) {
        ConnectivityManager cm = (ConnectivityManager) context
                .getSystemService(Context.CONNECTIVITY_SERVICE);

        NetworkInfo activeNetwork = cm.getActiveNetworkInfo();
        if (null != activeNetwork) {
            if (activeNetwork.getType() == ConnectivityManager.TYPE_WIFI)
                return TYPE_WIFI;

            if (activeNetwork.getType() == ConnectivityManager.TYPE_MOBILE)
                return TYPE_MOBILE;
        }
        return TYPE_NOT_CONNECTED;
    }

    public static String getConnectivityStatusString(Context context) {
        int conn = NetworkUtil.getConnectivityStatus(context);
        String status = null;
        if (conn == NetworkUtil.TYPE_WIFI) {
            status = "Wifi enabled";
        } else if (conn == NetworkUtil.TYPE_MOBILE) {
            status = "Mobile data enabled";
        } else if (conn == NetworkUtil.TYPE_NOT_CONNECTED) {
            status = "Not connected to Internet";
        }
        return status;
    }

    public static void getSSIDList(Context context) {
        /*
         * <uses-permission android:name="android.permission.ACCESS_WIFI_STATE"/>
         * Check this permission in manifest
         * */
        WifiManager wifiManager = (WifiManager) context
                .getSystemService(Context.WIFI_SERVICE);
        List<WifiConfiguration> configuredList = wifiManager.getConfiguredNetworks();
        List<String> ssidList = new ArrayList<>();

        for (WifiConfiguration config : configuredList) {
            ssidList.add(config.SSID);
        }
    }
}

File: app/src/main/java/com/surdroid/networkchangedetect/NetworkUtil.kt
package com.surdroid.networkchangedetect

import android.content.Context
import android.net.ConnectivityManager
import android.net.wifi.WifiManager
import com.surdroid.networkchangedetect.NetworkType.MOBILE
import com.surdroid.networkchangedetect.NetworkType.NOT_CONNECTED
import com.surdroid.networkchangedetect.NetworkType.WIFI


fun Context.getConnectivityStatus(): Int {
    val cm = getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
    return when (cm.activeNetworkInfo.type) {
        ConnectivityManager.TYPE_WIFI -> WIFI
        ConnectivityManager.TYPE_MOBILE -> MOBILE
        else -> NOT_CONNECTED
    }
}

fun Context.getConnectivityStatusString(): String? {
    return when (getConnectivityStatus()) {
        WIFI -> "Wifi enabled"
        MOBILE -> "Mobile data enabled"
        NOT_CONNECTED -> "Not connected to Internet"
        else -> null
    }
}

fun Context.getSSIDList(): List<String> {
    /*
     * <uses-permission android:name="android.permission.ACCESS_WIFI_STATE"/>
     * Check this permission in manifest
     * */
    val wifiManager = applicationContext.getSystemService(Context.WIFI_SERVICE) as WifiManager
    return wifiManager.configuredNetworks.map { it.SSID }
}

object NetworkType {
    var WIFI = 1
    var MOBILE = 2
    var NOT_CONNECTED = 0
}
